doc/manual/src/language/import-from-derivation.md

```diff
@@ -0,0 +1,94 @@
+# Import From Derivation
+
+Import From Derivation (IFD) is a Nix language pattern where an expression's value depends on the contents of a [store object] produced by a [derivation].
+It is characterised by the fact that the store object has to be [realise]d before the expression can be evaluated.
+
+[store object]: /glossary.md#gloss-store-object
+[derivation]: /glossary.md#gloss-derivation
+[realise]: /glossary.md#gloss-realise
+
+This has performance implications:
+Since evaluation is sequential, each required derivation will also be built sequentially.
+
+Passing a derivation `drv` to any built-in function that reads from the filesystem constitutes Import From Derivation:
```

Comment:
I'll use the "realize" word here as that is an orthogonal point.
```suggestion
Nix expressions can also depend on build results.

So far, we've only discussed Nix expressions that are either completely self contained, or depend on data that already exists (files on disk, something to be fetched, etc.).
But Nix expressions can also depend on build results, the [store object]s produced by [derivation]s., even if we haven't yet [realise]d that build result.
Evaluation will be paused if such a build result is needed, the build result realised, and then evaluation resumed.

[store object]: /glossary.md#gloss-store-object
[derivation]: /glossary.md#gloss-derivation
[realise]: /glossary.md#gloss-realise

This has performance implications:
Since evaluation is sequential, each required build result that the user doesn't already have in their store will also be built or downloaded sequentially.

Passing any expression that will evaluate to a [derived path] `drv^output` to any built-in function that reads from the filesystem constitutes Building-Dependent Evaluation:

[derived path]: /glossary.md#gloss-derived-path
```